src/api/user/RoleChannel.js

```diff
@@ -0,0 +1,52 @@
+import { BROADCAST_CHANNEL_NAME } from './constants';
+
+class RoleChannel {
+    constructor(openmct, channelName = BROADCAST_CHANNEL_NAME) {
+        this.openmct = openmct;
+        this.channelName = channelName;
+        this.roleChannel = undefined;
+    }
+
+    createRoleChannel() {
+        this.roleChannel = new BroadcastChannel(this.channelName);
+    }
+    subscribeToRole(cb) {
+        this.roleChannel.onmessage = (event => {
+            const role = event.data;
+            this.openmct.user.setActiveRole(role);
+            if (cb) {
+                cb(role);
+            }
+        });
+    }
+    unsubscribeToRole() {
+        this.roleChannel.close();
+    }
+    reconnect() {
+        this.roleChannel.close();
```

Comment:
Should this have a timeout here to prevent closing/creating channels too fast? We don't have a server here, so maybe this doesn't matter.